frappe/app.py

```diff
@@ -132,18 +130,17 @@ def application(request: Request):
 
 	except Exception as e:
 		response = handle_exception(e)
+		if db := getattr(frappe.local, "db", None):
+			db.rollback()
```

Comment:
alternatively we can just close here. `after_rollback` may not run in that case.